test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/SetTest.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.jsinterop
+
+import org.junit.Assert._
+import org.junit.Assume.assumeTrue
+import org.junit.{ BeforeClass, Test }
+import org.scalajs.testsuite.utils.Platform._
+
+import scala.scalajs.js
+
+object SetTest {
+  @BeforeClass
+  def assumeRuntimeSupportsSet(): Unit = {
+    assumeTrue("Assume Symbol and Set exists in Global",
+      jsSymbols && js.typeOf(js.constructorOf[js.Set[_]]) != "undefined")
+  }
+}
+
+class SetTest {
+
+  // scala.scalajs.js.Set
+
+  @Test def should_provide_clear(): Unit = {
+    val obj = js.Set("foo",  "bar")
+    assertTrue(obj.size == 2)
+    obj.clear()
+    assertTrue(obj.size == 0)
+  }
+
+  @Test def should_provide_an_iterator(): Unit = {
+    val obj = js.Set("foo", "bar", "babar")
+    var elems: List[String] = Nil
+    for (value <- obj) {
+      elems ::= value
+    }
+    assertEquals(3, elems.size)
+    assertTrue(elems.contains("foo"))
+    assertTrue(elems.contains("bar"))
+    assertTrue(elems.contains("babar"))
+  }
+
+  // scala.scalajs.js.JSConverters.JSRichGenSet
```

Comment:
Consider moving this comment inside `should_provide_toJSSet`. Currently it looks like a "section hear" and hence that it also applies to the following tests, which is not the case.